frontend/src/metabase/query_builder/components/filters/modals/BulkFilterModal/utils.ts

```diff
@@ -62,3 +68,37 @@ export const hasBackwardsArguments = (filter: Filter) => {
   const [lowerArgument, upperArgument] = filter.arguments();
   return lowerArgument > upperArgument;
 };
+
+export const searchByDimensionName = (
+  option: DimensionOption,
+  searchQuery: string,
+): boolean =>
+  option?.dimension
+    ?.displayName()
+    ?.toLowerCase()
+    ?.includes(searchQuery?.toLowerCase());
+
+export const searchBySegmentName = (
+  segment: SegmentOption,
+  searchQuery: string,
+): boolean =>
+  "segments".includes(searchQuery?.toLowerCase()) ||
```

Comment:
if there are segments defined for your table (your table probably doesn't have any), this will hit on the word segment in addition to the user-defined names of the segments

![Screen Shot 2022-07-11 at 12 20 49 PM](https://user-images.githubusercontent.com/30528226/178331694-0173ba4b-d622-4fbc-a0c9-23abf85da1ca.png)

Reply:
Hah, I see! Do you think the segment in the search should also say what table it's from like other searched attributes?

![image](https://user-images.githubusercontent.com/1937582/178481044-0d7e25c4-1776-4cd6-9605-2dbb05758e59.png)
vs
![image](https://user-images.githubusercontent.com/1937582/178481544-c595d910-ddb1-4424-a965-1e5dea0067f5.png)


Also, I'm not totally sure how segment searching works. I can select segment when viewing **People** and **Order** individually
![image](https://user-images.githubusercontent.com/1937582/178481755-7f8943d9-24f6-4c38-8342-5007d638bef9.png)

![image](https://user-images.githubusercontent.com/1937582/178481834-c1c5e285-9c5c-4f40-bece-dfe71c2d21c2.png)

But if I want to filter **User** (**Person** table) from **Order** then I won't see Person segment anymore.
![image](https://user-images.githubusercontent.com/1937582/178482018-4a01b347-d173-4780-99a0-fb9c69a85ca7.png)

Reply:
Having a constant on the left feels unusual. The way I read that is _"string 'segments' includes a search query"_, however, it's supposed to be something like _"search query contains the 'segments' string"_ I believe? I'd expect it to be something like `searchQuery?.toLowerCase().includes("segments")`. This is totally nit though

Reply:
@kulyk  except that won't match on the substring "segme" ??  i think the format is consistent with the format search `haystack.includes(needle)`

Reply:
Oh, that makes sense, thanks for explaining 👍

Reply:
@WiNloSt, segments should be table specific - one dropdown for a set of segments for each table. good catch on the table names. I'll make sure those show up.